README.md

```diff
@@ -1,2 +1,172 @@
+
+## Inputs
+### `suffixes`
+- **Description**: List of suffixes that test files should have, separated by commas.
+- **Required**: Yes
+- **Default**: `UnitTests,IntegrationTests`
+
+### `include_directories`
+- **Description**: List of directories to include in the pattern check. This input limits scanning to these directories only.
+- **Required**: No
+- **Default**: `src/test/`
+
+### `exclude_files`
+- **Description**: List of filenames to exclude from suffix checks, separated by commas.
+- **Required**: No
+- **Default**: None
+
+### `case_sensitivity`
+- **Description**: Determines if the filename check should be case-sensitive.
+- **Required**: No
+- **Default**: `true`
+
+### `logic`
+- **Description**: Switch logic from suffix (end of the filename) to contains (any part of the filename).
+- **Required**: No
+- **Default**: `false`
+
+### `report_format`
+- **Description**: Specifies the format of the output report. Options include console, csv, and json.
+- **Required**: No
+- **Default**: `console`
+
+### `verbose_logging`
+- **Description**: Enable verbose logging to provide detailed output during the action’s execution, aiding in troubleshooting and setup.
+- **Required**: No
+- **Default**: `false`
+
+### `fail_on_violations`
+- **Description**: Set to true to fail the action if any convention violations are detected. Set to false to continue without failure.
+- **Required**: No
+- **Default**: `false`
+
+## Outputs
+### `conventions_violations`
+- **Description**: Count test files not complying with the specified suffix conventions.
+
+### `report_path`
+- **Description**: Path to the generated report file.
+
+## Installation
+
+Clone the repository and navigate to the project directory:
+
+```bash
+git clone https://github.com/AbsaOSS/test-file-suffix-inspector.git
+cd test-file-suffix-inspector
+```
+
+Install the dependencies:
+```bash
+pip install -r requirements.txt
+```
+
+## Usage Example
+### Default
+```yaml
+name: Check Test File Naming Conventions
+on: [push]
+jobs:
+  check_naming:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Test File Suffix Inspector Default
+      id: scan-test-files
+      uses: AbsaOSS/test-file-suffix-inspector@v0.1.0
+      with:
+        suffixes: 'UnitTests,IntegrationTests'
+```
+
+### Full example
+```yaml
+name: Check Test File Naming Conventions
+on: [push]
+jobs:
+  check_naming:
+      runs-on: ubuntu-latest
+      steps:
+        - uses: actions/checkout@v2
+
+        - name: Test File Suffix Inspector Full Custom
+          id: scan-test-files
+          uses: AbsaOSS/test-file-suffix-inspector@v0.1.0
+          with:
+            suffixes: 'UnitTests,IntegrationTests'
+            include_directories: 'src/test/'
```

Comment:
How does this work? Does it do `src/test` per module so it is always like `{module}/src/test/*`. 
Does it use glob patterns?

Reply:
The action uses the project's current working dir as a starting point. Then, it scans for all folders on the current level, removes dir exclusions, excludes by file exclusions, and checks for file patterns.

The logic does not expect any project structure. It is oriented only on folders and files.

Reply:
Ok, I think I get it, but I don't believe an include should be optional.

Also, do we support some patterns like `*`? Both in include or suffix definition.

Reply:
No * patterns are there in this version.